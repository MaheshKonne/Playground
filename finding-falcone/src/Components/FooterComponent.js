import React, { Fragment } from 'react';
import theme from "../theme";

const FooterComponent = () => (
    <Fragment>
        <footer>
            <div>
                <a href='www.geektrust.in/finding-falcone'>
                    Coding problem - www.geektrust.in/finding-falcone
                </a>
            </div>
        </footer>
        <style>
            {`
            footer {
                height: 70px;
                display: flex;
                justify-content: center;
                align-items: center;
                background-color: ${theme.headerBackground};
                position: fixed;
                width: 100%;
                bottom: 0;
            }
            a {
                text-decoration: none;
                color: black;
            }
            `}
        </style>
    </Fragment>
)

export default FooterComponent;