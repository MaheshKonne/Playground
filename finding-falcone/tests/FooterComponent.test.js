import React from 'react';
import FooterComponent from '../src/Components/FooterComponent';
import {shallow} from "enzyme";

describe('<FooterComponent />', () => {
    const component = shallow(<FooterComponent/>);

    it('should match with snapshot', function () {
       expect(component).toMatchSnapshot();
    });
});