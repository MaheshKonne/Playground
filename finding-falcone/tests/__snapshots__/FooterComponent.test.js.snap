// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FooterComponent /> should match with snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FooterComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <footer>
          <div>
            <a
              href="www.geektrust.in/finding-falcone"
            >
              Coding problem - www.geektrust.in/finding-falcone
            </a>
          </div>
        </footer>,
        <style>
          
            footer {
                height: 70px;
                display: flex;
                justify-content: center;
                align-items: center;
                background-color: rgba(200,200,200,0.5);
                position: fixed;
                width: 100%;
                bottom: 0;
            }
            a {
                text-decoration: none;
                color: black;
            }
            
        </style>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            <a
              href="www.geektrust.in/finding-falcone"
            >
              Coding problem - www.geektrust.in/finding-falcone
            </a>
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="www.geektrust.in/finding-falcone"
            >
              Coding problem - www.geektrust.in/finding-falcone
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Coding problem - www.geektrust.in/finding-falcone",
              "href": "www.geektrust.in/finding-falcone",
            },
            "ref": null,
            "rendered": "Coding problem - www.geektrust.in/finding-falcone",
            "type": "a",
          },
          "type": "div",
        },
        "type": "footer",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "
            footer {
                height: 70px;
                display: flex;
                justify-content: center;
                align-items: center;
                background-color: rgba(200,200,200,0.5);
                position: fixed;
                width: 100%;
                bottom: 0;
            }
            a {
                text-decoration: none;
                color: black;
            }
            ",
        },
        "ref": null,
        "rendered": "
            footer {
                height: 70px;
                display: flex;
                justify-content: center;
                align-items: center;
                background-color: rgba(200,200,200,0.5);
                position: fixed;
                width: 100%;
                bottom: 0;
            }
            a {
                text-decoration: none;
                color: black;
            }
            ",
        "type": "style",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <footer>
            <div>
              <a
                href="www.geektrust.in/finding-falcone"
              >
                Coding problem - www.geektrust.in/finding-falcone
              </a>
            </div>
          </footer>,
          <style>
            
            footer {
                height: 70px;
                display: flex;
                justify-content: center;
                align-items: center;
                background-color: rgba(200,200,200,0.5);
                position: fixed;
                width: 100%;
                bottom: 0;
            }
            a {
                text-decoration: none;
                color: black;
            }
            
          </style>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div>
              <a
                href="www.geektrust.in/finding-falcone"
              >
                Coding problem - www.geektrust.in/finding-falcone
              </a>
            </div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="www.geektrust.in/finding-falcone"
              >
                Coding problem - www.geektrust.in/finding-falcone
              </a>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Coding problem - www.geektrust.in/finding-falcone",
                "href": "www.geektrust.in/finding-falcone",
              },
              "ref": null,
              "rendered": "Coding problem - www.geektrust.in/finding-falcone",
              "type": "a",
            },
            "type": "div",
          },
          "type": "footer",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "
            footer {
                height: 70px;
                display: flex;
                justify-content: center;
                align-items: center;
                background-color: rgba(200,200,200,0.5);
                position: fixed;
                width: 100%;
                bottom: 0;
            }
            a {
                text-decoration: none;
                color: black;
            }
            ",
          },
          "ref": null,
          "rendered": "
            footer {
                height: 70px;
                display: flex;
                justify-content: center;
                align-items: center;
                background-color: rgba(200,200,200,0.5);
                position: fixed;
                width: 100%;
                bottom: 0;
            }
            a {
                text-decoration: none;
                color: black;
            }
            ",
          "type": "style",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
